### PR TITLE
Added <properties> array built from meta added per test

### PR DESCRIPTION
**Pull request by tamusjroyce**
Added <properties> array built from meta added per test

I have created a hook that displays timings for every endpoint that is hit:

```
// npx tsc to compile this to js
import { RequestHook } from 'testcafe';

export class TimingRequestHook extends RequestHook {
    public hookTest = (fn: (i: any) => Promise<any> | Promise<void>) => {
        const flushing = this.flush;
        return async (t: any) => {
            try {
                return await fn(t);
            } finally {
                flushing(t.testRun?.test ?? t.test ?? t);
            }
        }
    };

    public requestMap: Map<string, { time: number, request: any }>;
    public queuedMap: { [key: string]: string };

    constructor (requestFilterRules: any, responseEventConfigureOpts: any = {}) {
        super(requestFilterRules, responseEventConfigureOpts);
        this.requestMap = new Map<string, { time: number, request: any }>();
        this.queuedMap = {};
    }

    public onRequest (event: any) {
        // ...
        console.log(event._requestInfo.url, '->', event._requestInfo.requestId, '->', event);
        this.requestMap.set(event._requestInfo.requestId, { time: new Date().getTime(), request: event });
    }

    public onResponse (event: any) {
        // ...
        const id = event && event.requestId;
        if (!this.requestMap.has(id)) {
            return;
        }
        const request = this.requestMap.get(id);
        if (!request) {
            console.log('response request not found for requestId:', id);
            return;
        }
        const time = ((new Date().getTime()) - request.time) / 1000.0;
        console.log('response time for ', request.request._requestInfo.url, ':', time, 's');

        this.queuedMap[request.request._requestInfo.url] = String(time);
        this.requestMap.delete(id);
    }

    public flush = (meta: any): any => {
        const metaObject = meta?.meta ?? meta;
        if (typeof(metaObject) === 'function') {
            metaObject(this.queuedMap);
        } else if (typeof(metaObject) === 'object') {
            Object.keys(metaObject).forEach(key => {
                delete metaObject[key];
            });
            Object.keys(this.queuedMap).forEach(key => {
                metaObject[key] = this.queuedMap[key];
            });
        }
        return meta;
    };
}
```
I also plan on expanding this to use quicktype.io to generate OpenAPI spec and suggest typescript interfaces.

Usage of hook above:

```
...

import { TimingRequestHook } from '../../framework/helpers/px-request-hook.helper';
const hook = new TimingRequestHook ([/.*[yoursite].*ping.*/, /.*[yoursite].*health.*/]);

test('Your Test Description', hook.hookTest(async (t) => {
  await welcome.signIn(setEmail(), setPassword());
  await account.selectAccount();
  await home.closePopup();
  await t
    .expect((await header.mainTitle.innerText).trim())
    .eql('your site page');
}));
...
```

So you have to wrap your test around hook.hookTest(); There may be a better way. But quick and easy.

### CODE REVIEW

**Comment by tamusjroyce** [comment]
The above code is to support my new hook. It does not include this new hook.

**Comment by tamusjroyce** [comment]
```
<?xml version="1.0" encoding="UTF-8" ?>
<test-run id="2" name="TestCafe Tests" start-time="2022-11-11T07:05:38.980Z" end-time="2022-11-11T07:06:23.918Z">
    <test-suite type="Assembly" name="Your Tests" start-time="2022-11-11T07:05:38.984Z" end-time="2022-11-11T07:06:23.915Z">
        <test-case name="Your test description" fullname="Your test description - everything that can describe your tests" result="Passed" duration="44.908">
            <properties>
                    <property name="https://[yoursite]/api/yourapi/1" value="0.5" />
                    <property name="https://[yoursite]/api/yourapi/2" value="0.74" />
                    <property name="https://[yoursite]/api/yourapi/3" value="2.088" />
                    <property name="https://[yoursite]/api/yourapi/4" value="0.201" />
                    <property name="https://[yoursite]/api/yourapi/5" value="0.25" />
                    <property name="https://[yoursite]/api/yourapi/6" value="1.563" />
            </properties>
            <failure>
                <message>
                    <![CDATA[Run 1: Passed
Run W2JP8aYmq: Passed]]>
                </message>
                <stack-trace>
                    <![CDATA[]]>
                </stack-trace>
            </failure>
        </test-case>
    </test-suite>
</test-run>
```

Example xml with <properties><property ... /></properties> added

File: src/template.handlebars [hunk 1]
```diff
@@ -4,6 +4,13 @@
     <test-suite type="Assembly" name="{{name}}" start-time="{{startTime}}" end-time="{{endTime}}">
         {{#testCases}}
         <test-case name="{{name}}" fullname="{{../name}} - {{name}}" result="{{result}}" duration="{{duration}}">
+            {{#if hasMetaData}}
+            <properties>
+                {{#each meta}}
+                    <property name="{{@key}}" value="{{this}}" />
```
Comment (tamusjroyce) [inline src/template.handlebars:10]: Removing 4 spaces before <property name=...> would make the report.xml a little look better. But ok as is.

**Comment by NickLargen** [comment]
I'll take a look at this within the next few days.

**Comment by NickLargen** [comment]
Thanks, added along with fixture metadata to release 3.1.

**Pull request closed without merging**